### PR TITLE
Make sure manifest.json contains AUTOUPDATE_VERSION (if specified)

### PR DESCRIPTION
**Pull request by npvn**
Fixes https://github.com/meteor/meteor/issues/8479

### CODE REVIEW

File: packages/webapp/webapp_server.js [hunk 1]
```diff
@@ -500,8 +500,12 @@ var runWebAppServer = function () {
         var program = {
           format: "web-program-pre1",
           manifest: manifest,
-          version: WebAppHashing.calculateClientHash(manifest, null, _.pick(
-            __meteor_runtime_config__, 'PUBLIC_SETTINGS')),
+          version: process.env.AUTOUPDATE_VERSION ||
```
Comment (benjamn) [inline packages/webapp/webapp_server.js]: Can we call this `METEOR_AUTOUPDATE_VERSION`?

Comment (benjamn) [inline packages/webapp/webapp_server.js]: Ah, @abernix tells me it's not a new variable, so we should probably keep it as-is.

**Comment by abernix** [comment]
@martijnwalraven This _might_ not even be Cordova-only, but you might be more familiar with the implementation here and this hasn't changed in so long.  There does seem to actually be a bug here when using `AUTOUPDATE_VERSION` (as demonstrated in the linked issue, #8479), and this does fix the problem, but I'm not 100% certain this is the correct solution for it given other "autoupdate" logic, such as that in [`WebAppHashing.calculateClientHash`](https://github.com/meteor/meteor/blob/8b846ff2509493a2af49505201d143d99c7155a0/packages/webapp-hashing/webapp-hashing.js#L18-L26) (which seems like it should be a self-contained function for determining this).

This does seem like very similar treatment to https://github.com/meteor/meteor/blob/devel/tools/cordova/builder.js#L411 and https://github.com/meteor/meteor/blob/devel/packages/autoupdate/autoupdate_server.js#L76-L84 but I can't help but feel like this might be a bug somewhere else.

If you're not sure, I can dig into it more!

**Comment by martijnwalraven** [comment]
The fix looks good to me, although I have to admit it's been a while since I looked at the autoupdate logic.

> I'm not 100% certain this is the correct solution for it given other "autoupdate" logic, such as that in [`WebAppHashing.calculateClientHash`](https://github.com/meteor/meteor/blob/8b846ff2509493a2af49505201d143d99c7155a0/packages/webapp-hashing/webapp-hashing.js#L18-L26) (which seems like it should be a self-contained function for determining this).

As far as I know, `AUTOUPDATE_VERSION` is meant as a way to override `WebAppHashing.calculateClientHash`, so that makes sense I think.

**Comment by abernix** [comment]
Thanks, @martijnwalraven.  It certainly does solve the problem, and if it was meant to override the `calculateClientHash` (rather than being included or calculated as part of it), then this LGTM.

@npvm Do you know if this problem exists in a non-Cordova project for an app using `AUTOUPDATE_VERSION`?  If so, this might be nice to test along the lines of [this test](https://github.com/meteor/meteor/blob/devel/tools/tests/hot-code-push.js#L94-L117).  Do you think that's possible?

**Comment by npvn** [comment]
Thanks everyone for taking a look at this.

@abernix I can't reproduce this issue with a non-Cordova client. The reason is properly that non-Cordova clients don't read the version from `__cordova/manifest.json`.

**Comment by abernix** [comment]
This should be available for testing in Meteor 1.4.4-rc.4.  Please try it out and provide feedback! :smile:

    meteor update --release 1.4.4-rc.4

**Comment by npvn** [comment]
The fix is working great in `1.4.4-rc.4`. Thanks everyone!

**Pull request merged**